src/main/resources/com/google/api/codegen/nodejs/index.snip

```diff
@@ -0,0 +1,43 @@
+@extends "nodejs/common.snip"
+
+@snippet generate(index)
+  /*
+   {@copyright()}
+   */
+  'use strict';
+
+  @join require : index.requireViews
+    var {@require.clientName} = require('./{@require.fileName}');
+  @end
+  var gax = require('google-gax');
+  var extend = require('extend');
+  @if index.hasMultipleServices
+    var union = require('lodash.union');
+  @end
+
+  function {@index.apiVersion}(options) {
+    options = extend({
+      scopes: {@index.apiVersion}.ALL_SCOPES
+    }, options);
+    var gaxGrpc = gax.grpc(options);
+    var result = {};
+    @join require : index.requireViews
+      extend(result, {@require.clientName}(gaxGrpc));
+    @end
+    return result;
+  }
+
+  {@index.apiVersion}.SERVICE_ADDRESS = {@index.primaryService.clientName}.SERVICE_ADDRESS;
+  @if index.hasMultipleServices
+    {@index.apiVersion}.ALL_SCOPES = union(
+      @join require : index.requireViews on ", ".add(BREAK)
+        {@require.clientName}.ALL_SCOPES
+      @end
+    );
+  @else
+    {@index.apiVersion}.ALL_SCOPES = {@index.primaryService.clientName}.ALL_SCOPES;
```

Comment:
You don't have to do that in this PR, but I'm wondering if it's possible to define the list of all scopes here and remove `ALL_SCOPES` in each service .js file.